# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieServiceChild.cpp
# Commit: 302d12c12083
# Full Hash: 302d12c12083571db812384c7ecc8ab5d466d517
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-07 21:34:02
# Regressor Bug: 1839918
# File Overlap Count: 1
# Description:
#   Bug 1839918 - Use PHttpChannel for SetCookies calls from child channels. r=bvandersloot,necko-reviewers,cookie-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D186340
# ==============================================================================

diff -r 750e0796c1ad -r 302d12c12083 netwerk/cookie/CookieServiceChild.cpp
--- a/netwerk/cookie/CookieServiceChild.cpp	Thu Sep 07 08:25:55 2023 +0000
+++ b/netwerk/cookie/CookieServiceChild.cpp	Thu Sep 07 08:25:55 2023 +0000
@@ -8,6 +8,7 @@
 #include "CookieLogging.h"
 #include "CookieService.h"
 #include "mozilla/net/CookieServiceChild.h"
+#include "mozilla/net/HttpChannelChild.h"
 #include "mozilla/net/NeckoChannelParams.h"
 #include "mozilla/LoadInfo.h"
 #include "mozilla/BasePrincipal.h"
@@ -28,6 +29,7 @@
 #include "nsIURI.h"
 #include "nsIPrefBranch.h"
 #include "nsIWebProgressListener.h"
+#include "nsQueryObject.h"
 #include "nsServiceManagerUtils.h"
 #include "mozilla/Telemetry.h"
 #include "mozilla/TimeStamp.h"
@@ -571,7 +573,10 @@
 
   // Asynchronously call the parent.
   if (CanSend() && !cookiesToSend.IsEmpty()) {
-    SendSetCookies(baseDomain, attrs, aHostURI, true, cookiesToSend);
+    RefPtr<HttpChannelChild> httpChannelChild = do_QueryObject(aChannel);
+    MOZ_ASSERT(httpChannelChild);
+    httpChannelChild->SendSetCookies(baseDomain, attrs, aHostURI, true,
+                                     cookiesToSend);
   }
 
   return NS_OK;